CMakeLists.txt

```diff
@@ -1,49 +1,109 @@
 cmake_minimum_required(VERSION 2.8)
-project(x86energy_plugin)
+project(x86_energy_plugin)
 
-option(BACKEND_SCOREP "Build plugin using scorep(ON) or vampirtrace(OFF)" ON)
+# Intialize git submodules if not done already
+file(GLOB SUBMODULE_FILES "common/*")
+list(LENGTH SUBMODULE_FILES COUNT_COMMON)
+file(GLOB SUBMODULE_FILES "x86_energy/*")
+list(LENGTH SUBMODULE_FILES COUNT_X86_ENERGY)
+file(GLOB SUBMODULE_FILES "scorep_cxx_wrapper/*")
+list(LENGTH SUBMODULE_FILES COUNT_SCOREP)
+if (${COUNT_COMMON} EQUAL 0 OR ${COUNT_SCOREP} EQUAL 0 OR ${COUNT_X86_ENERGY} EQUAL 0 )
+    message(STATUS "Initializing git submodule")
+    execute_process(COMMAND "git" "submodule" "init" WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}")
+    execute_process(COMMAND "git" "submodule" "update" WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}")
```

Comment:
That is a valid point. But, then you pull a new version it won't update the submodule. I ran into this on another project's Jenkins build. Plus, how often do you have local changes in a submodule?

Maybe we add something like this:
```
option(GIT_UPDATE_SUBMODULES "Automatically update git submodules during CMake run" ON)

if(GIT_UPDATE_SUBMODULES)
    execute_process(COMMAND "git" "submodule" "update" WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}")
endif()
```

This way, you can conveniently turn it off, then you have local changes.